packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelFactoryProvider.cs

```diff
@@ -126,11 +126,11 @@ private IReadOnlyList<MethodBodyStatement> GetCollectionInitialization(MethodSig
             {
                 if (param.Type.IsList)
                 {
-                    statements.Add(param.Assign(New.Instance(new CSharpType(typeof(List<>), param.Type.Arguments))).Terminate());
+                    statements.Add(param.Assign(New.Instance(CodeModelPlugin.Instance.TypeFactory.ListInitializationType.MakeGenericType(param.Type.Arguments))).Terminate());
```

Comment:
can we just do `param.Type.PropertyInitializationType` ?

Reply:
The ListInitializationType method is on the TypeFactory so I accessed it through `CodeModelPlugin.Instance.TypeFactory.ListInitializationType.`. I don't think we can access the ListInitializationType from param.Type.PropertyInitializationType? I think we need the one with the ChangeTrackingList

Reply:
`PropertyInitializationType` is a public API on `CsharpType`, I think we should do this:
```suggestion
                    statements.Add(param.Assign(New.Instance(param.Type.PropertyInitializationType)).Terminate());
```
in this way you do not even need to worry about the generic arguments.
Also it makes sense here - in the model factory method, we are calling the "internal ctor" whose parameters are actually their properties directly, and we need to "initialize" those properties, and hence we are using the "property initialization type" here, instead of the type of the parameter which is input variant of the property type.

Reply:
That makes sense to me! Thanks Dapeng